openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/core/CoreNodeContractor.java

```diff
@@ -0,0 +1,457 @@
+                   TraversalMode traversalMode) {
+        if (traversalMode.isEdgeBased()) {
+            throw new IllegalArgumentException("Contraction Hierarchies only support node based traversal so far, given: " + traversalMode);
+        }
+        // todo: it would be nice to check if ghStorage is frozen here
+        this.ghStorage = ghStorage;
+        this.prepareGraph = prepareGraph;
+        this.prepareWeighting = new PreparationWeighting(weighting);
+        this.traversalMode = traversalMode;
+        originalEdges = dir.find("original_edges_" + AbstractWeighting.weightingToFileName(weighting));
+        originalEdges.create(1000);
+    }
+
+    void initFromGraph() {
+        // todo: do we really need this method ? the problem is that ghStorage/prepareGraph can potentially be modified
+        // between the constructor call and contractNode,calcShortcutCount etc. ...
+        maxLevel = prepareGraph.getNodes() + 1;
+        maxEdgesCount = ghStorage.getAllEdges().getMaxId();
+        ignoreNodeFilterSequence = new IgnoreNodeFilterSequence(prepareGraph, maxLevel);
+        ignoreNodeFilterSequence.add(restrictionFilter);
+        FlagEncoder prepareFlagEncoder = prepareWeighting.getFlagEncoder();
+        vehicleInExplorer = prepareGraph.createEdgeExplorer(new DefaultEdgeFilter(prepareFlagEncoder, true, false));
+        vehicleOutExplorer = prepareGraph.createEdgeExplorer(new DefaultEdgeFilter(prepareFlagEncoder, false, true));
+        prepareAlgo = new DijkstraOneToMany(prepareGraph, prepareWeighting, traversalMode);
+    }
+
+    void close() {
+        prepareAlgo.close();
+        originalEdges.close();
+    }
+
+    void setRestrictionFilter(EdgeFilter filter){
+        this.restrictionFilter = filter;
+    }
+
+    void setMaxVisitedNodes(int maxVisitedNodes) {
+        this.maxVisitedNodes = maxVisitedNodes;
+    }
+
+    long contractNode(int node) {
+        shortcuts.clear();
+        long degree = findShortcuts(addScHandler.setNode(node));
+        addedShortcutsCount += addShortcuts(shortcuts.keySet());
+        return degree;
+    }
+
+    CalcShortcutsResult calcShortcutCount(int node) {
+        findShortcuts(calcScHandler.setNode(node));
+        return calcScHandler.calcShortcutsResult;
+    }
+
+    /**
+     * Searches for shortcuts and calls the given handler on each shortcut that is found. The graph is not directly
+     * changed by this method.
+     * Returns the 'degree' of the handler's node (disregarding edges from/to already contracted nodes). Note that
+     * here the degree is not the total number of adjacent edges, but only the number of incoming edges
+     */
+    private long findShortcuts(ShortcutHandler sch) {
+        long tmpDegreeCounter = 0;
+        EdgeIterator incomingEdges = vehicleInExplorer.setBaseNode(sch.getNode());
+        // collect outgoing nodes (goal-nodes) only once
+        while (incomingEdges.next()) {
+            int u_fromNode = incomingEdges.getAdjNode();
```

Comment:
We shouldn't use underscores in variable names

Reply:
This method uses the original GH's notation, where `v`, `u`, and `w` denote vertices. I don't immediately see how the naming could be improved here without sacrificing code readability, so I would leave it as is for now.